File: hospital-fe-main/src/app/modules/hospital/model/specializationSearchAppointmentDTO.ts
import {DateRange} from "./date-range.model";

enum Priority {
  octorPriority, TimePriority
}
enum Specialization{
  Dermatolog,
  Opsta_praksa,
  Kardiolog,
  Fizijatar
}

export class SpecializationSearchAppointmentDTO {
  priority: Priority;
  specialization: Specialization;
  doctorId : string = '';
  range!: DateRange;

  public constructor(doctor: string = "", rangee: DateRange) {
    this.doctorId = doctor;
    this.range = rangee;
  }
}

File: hospital-be-main/src/HospitalLibrary/Core/DTO/SpecializationSearchAppointmentDTO.cs
using HospitalLibrary.Core.Enum;
using HospitalLibrary.Core.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HospitalLibrary.Core.DTO
{
    public class SpecializationSearchAppointmentDTO
    {
        public Priority Priority { get; set; }
        public Specialization Specialization { get; set; }
        public Guid DoctorId { get; set; }
        public DateRange Range { get; set; }
    }
}
